src/pybind/mgr/pg_autoscaler/module.py

```diff
@@ -383,6 +387,45 @@ def _maybe_adjust(self):
                     'var': 'pg_num',
                     'val': str(p['pg_num_final'])
                 })
+                
+                # Create new event for each pool
+                # and update existing events
+                # Call Progress Module to create progress event
+
+                if pool_id not in self._event:
+                    self._event[pool_id] = str(uuid.uuid4())
+                    if p['pg_num_target'] < p['pg_num_final']:
+                        self.remote('progress', 'update', self._event[pool_id], 
+                                ev_msg=" PgAutoscaler growing PGs in pool: {0}".format(
+                                    pool_id), 
+                                ev_progress=0.0, 
+                                refs=[("pool", int(pool_id))])
+                    else:
+                        self.remote('progress', 'update', self._event[pool_id], 
+                                ev_msg=" PgAutoscaler shrinking PGs in pool: {0}".format(
+                                    pool_id), 
+                                ev_progress=0.0, 
+                                refs=[("pool", int(pool_id))])
+
+                else:
+                    new_progress = p['pg_num_target']/p['pg_num_final']
```

Comment:
I think this path is only taken once, when we set ``pg_num_target``.  Isn't there an update method of some sort to refresh progress (that would need to compare pg_num with pg_num_target and the initial pg_num)?

Reply:
Hi Sage, 
So from my understanding (could be wrong) is that p['pg_num_final'] is the current pg_num in the pool and p['p_num_target'] is the target PG the pool is trying to reach. As to how it refreshes the progress, I think the function _maybe_adjust get called every 1 min as specified by the 'sleep interval', and calls the function get_osdmap() to get a new map every time and use it to call get_pools_by_name to get new information about the pools. Therefore, I feel like it is already updating at that point

Reply:
``pg_num_target`` is what ceph currently wants ``pg_num`` to be.
``pg_num`` is the current, real pg count for the pool. The mgr will move ``pg_num`` toward ``pg_num_target`` for you... this is what should update the progress.
``pg_num_final`` is what the pg_autoscaler module decided it wants ``pg_num_target`` to be.

If I'm reading the code correctly, the ``would_adjust`` will be true when we first make the adjustment to ``pg_num_target``.  But early in ``_maybe_adjust()`` you have

```
            if not p['would_adjust']:
                continue
```
So you want to *create* the progress event there, but that's not the right place to update progress.

Instead, I think you want a separate helper like ``_update_progress_events()`` or something like that.  Loop over the progress event objects you have in the active list, and compare the latest ``pg_num`` to ``pg_num_target`` and the starting ``pg_num`` value.  I think to do this you need to create a class like ``PgAdjustmentProgress`` that keeps the initial and target ``pg_num`` values...